src/script/components/list/GroupedConversations.tsx

```diff
@@ -0,0 +1,167 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React from 'react';
+
+import type {ConversationRepository} from '../../conversation/ConversationRepository';
+import type {Conversation} from '../../entity/Conversation';
+import {
+  ConversationLabel,
+  createLabel,
+  createLabelFavorites,
+  createLabelGroups,
+  createLabelPeople,
+} from '../../conversation/ConversationLabelRepository';
+import {generateConversationUrl} from '../../router/routeGenerator';
+import {createNavigate} from '../../router/routerBindings';
+
+import './GroupedConversationHeader';
+import GroupedConversationHeader from './GroupedConversationHeader';
+import ConversationListCell from './ConversationListCell';
+import {ListViewModel} from 'src/script/view_model/ListViewModel';
+import {registerReactComponent} from 'Util/ComponentUtil';
+
+export interface GroupedConversationsProps {
+  conversationRepository: ConversationRepository;
+  expandedFolders: string[];
+  hasJoinableCall: (conversationId: string) => boolean;
+  isSelectedConversation: (conversationEntity: Conversation) => boolean;
+  isVisibleFunc: (top: number, bottom: number) => boolean;
+  listViewModel: ListViewModel;
+  onJoinCall: (conversationEntity: Conversation) => void;
+  setExpandedFolders: (folders: string[]) => void;
+}
+
+const GroupedConversations: React.FC<GroupedConversationsProps> = ({
+  conversationRepository,
+  expandedFolders,
+  hasJoinableCall,
+  isSelectedConversation,
+  isVisibleFunc,
+  listViewModel,
+  onJoinCall,
+  setExpandedFolders,
+}) => {
+  const {conversationLabelRepository} = conversationRepository;
+  const makeOnClick = (conversationId: string, domain: string | null) =>
+    createNavigate(generateConversationUrl(conversationId, domain));
+
+  const folders = (() => {
+    const folders: ConversationLabel[] = [];
+
+    const favorites = conversationLabelRepository.getFavorites();
+    if (favorites.length) {
+      folders.push(createLabelFavorites(favorites));
+    }
+
+    const groups = conversationLabelRepository.getGroupsWithoutLabel();
+    if (groups.length) {
+      folders.push(createLabelGroups(groups));
+    }
+
+    const contacts = conversationLabelRepository.getContactsWithoutLabel();
+    if (contacts.length) {
+      folders.push(createLabelPeople(contacts));
+    }
+
+    const custom = conversationLabelRepository
+      .getLabels()
+      .map(label => createLabel(label.name, conversationLabelRepository.getLabelConversations(label), label.id))
+      .filter(({conversations}) => !!conversations().length);
+    folders.push(...custom);
+
+    return folders;
+  })();
+
+  const isExpanded = (folderId: string): boolean => expandedFolders.includes(folderId);
+  const toggle = (folderId: string): void => {
+    if (isExpanded(folderId)) {
+      setExpandedFolders(expandedFolders.filter(folder => folder !== folderId));
```

Comment:
As these changes rely on the previous state, a setter function would be more fitting: 
```suggestion
      setExpandedFolders(expanded => expanded.filter(folder => folder !== folderId));
```